osu.Game.Rulesets.Catch/Skinning/ICatcherSprite.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Textures;
+
+namespace osu.Game.Rulesets.Catch.Skinning
+{
+    public interface ICatcherSprite
+    {
+        Texture CurrentTexture { get; }
```

Comment:
This looks like quite the leaky abstraction (a catcher sprite exposing raw textures?) Something like

```csharp
void ApplyStateTo(CatcherTrailSprite trailSprite)
```

would fare better I think.

Reply:
Since this is such a local implementation I don't think it's too much of a concern.